### PR TITLE
Add Photon Realtime tutorials page

### PR DESCRIPTION
**Pull request by yushimatenjin**
I have written about how to combine the JavaScript SDK with PlayCanvas.

### Links
Photon
https://www.photonengine.com/

### External projects
**1. A simple project for the tutorial .**
~~Project URL: https://playcanvas.com/project/928285/~~
~~Publish URL: https://playcanv.as/e/p/sLGr3Ql2/~~

Project URL: https://playcanvas.com/project/954410/
Publish URL: https://playcanv.as/p/wDUPyoOS/

**2. The full project, including matchmaking.**
Project URL: https://playcanvas.com/project/926999/
Publish URL: https://playcanv.as/e/p/dvJYXs1Y/

### Capture
localhost_51000_en_tutorials_real-time-multiplayer-photon_ (3)

### CODE REVIEW

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+
+- [API Reference][14]
+- [Photon Engine - glossary][15]
+
+## Using Photon with PlayCanvas
+
+### 1. Using the LoadBalancing class in the SDK
+
+Develop using a class that extends **LoadBalancingClient**.
+Please create **photon-loadbalancing.js**.
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:119]:
```suggestion
To get started, we need to create a class that extends **LoadBalancingClient**. Please create a new script asset named **photon-loadbalancing.js** and add the following code:
```

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+
+        // pc.Application
+        this.app = props.app;
+    }
+}
+```
+
+### 2. Instantiate classes from PlayCanvas to use Photon
+
+Create a **photon-loadbalancing-playcanvas.js** to the project to initialize Photon.
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:139]:
```suggestion
Create a script asset named **photon-loadbalancing-playcanvas.js** to the project to initialize Photon.
```

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+    { "USA, West San José": "usw" },
+  ],
+});
+```
+
+### 3. Set Script for Root entity
+
+Create a new **photon-loadbalancing-playcanvas.js** and attach it to the root entity.
+
+![Root Entity - Inspector][16]
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:197]: Missing image from preview

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+    { "Turkey Istanbul": "tr" },
+    { "USA, East Washington": "us" },
+    { "USA, West San José": "usw" },
+  ],
+});
+```
+
+### 3. Set Script for Root entity
+
+Create a new **photon-loadbalancing-playcanvas.js** and attach it to the root entity.
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:195]:
```suggestion
Create a new script asset **photon-loadbalancing-playcanvas.js** and attach it to the Root entity in the Editor.
```

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+
+![Root Entity - Inspector][16]
+
+### 4. Paste AppId into the script attribute.
+
+Enter AppId as a script attribute.
+
+![Script Attributes][17]
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:203]: Missing image from preview

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+  wss: this.wss ? 1 : 0,
+  appId: this.appId,
+  appVersion: this.appVersion,
+  app: this.app,
+});
+```
+
+- **wss** Secure connection via WebSocket.
+- **appId** The value application identifier.
+- **appVersion** Game version. Can be a separate Virtual Application.
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:216]: What's a virtual application?

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 7]
```diff
@@ -0,0 +1,528 @@
...
+    }
+  }
+}
+
+pc.registerScript(Player);
+```
+
+- **this.app.fire** [communication][22] between scripts.
+
+### If successful
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:497]:
```suggestion
### Done!
```

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 8]
```diff
@@ -0,0 +1,528 @@
...
+
+You can play multiplayer in Photon.
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:499]:
```suggestion
You can now play multiplayer in Photon!
```

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 9]
```diff
@@ -0,0 +1,528 @@
...
+
+![Project][23]
+
+Now you can create a room using Photon and synchronize the location information of the characters.
```
Comment (yaustar) [inline content/en/tutorials/real-time-multiplayer-photon.md:503]:
```suggestion
You can create a room using Photon and synchronize the positions of players with each other.
```

**Comment by yushimatenjin** [comment]
@yaustar 
Thank you so much for your review !
I appreciate the help with this. 
I’ll revise it.

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 10]
```diff
@@ -0,0 +1,528 @@
...
+
+### 3. Click download SDK
+
+![Download SDK][10]
+
+### 4. Unzip the SDK
+
+![Unzip SDK][11]
+
+The SDK will be downloaded in ZIP format, unzip it: `photon-javascript-sdk_v4-1-1-0` → `lib` → **`Photon-Javascript_Emscripten_SDK.min.js`**.
```
[the remark this answers is not in the document]
Comment (yushimatenjin) [reply content/en/tutorials/real-time-multiplayer-photon.md:84]: Sorry, there was a problem with the way I took the screenshot in Chrome.
I took it again and it worked.

File: content/en/tutorials/real-time-multiplayer-photon.md [hunk 11]
```diff
@@ -0,0 +1,528 @@
...
+- How to use Photon SDK
+- Multiplayer implementation
+
+### Tutorial Project
+
+<iframe loading="lazy" src="https://playcanv.as/e/p/sLGr3Ql2/" title="Real-time Multiplayer with Photon - a simple project"></iframe>
+
+*Arrow keys to move.*
+
+You can find the [tutorial project here][2].
```
[the remark this answers is not in the document]
Comment (yushimatenjin) [reply content/en/tutorials/real-time-multiplayer-photon.md:27]: Thank you! I will remove this one.

**Comment by yushimatenjin** [comment]
@yaustar I am sorry for being late to contact you.

I fixed it.
- Changed from Class to Prototype implementation
- Changed Tutorial
- Retaking screenshots
- Fixed many mistakes

### Capture
-----

**Comment by yaustar** [comment]
Looks good on my side.

The game/published build still doesn't create a room for me. Is something extra needed?

https://user-images.githubusercontent.com/16639049/177743157-02efcf1a-ae50-466a-b7df-951cc5be0745.mp4

@willeastcott Can you do a final pass please?

**Comment by yushimatenjin** [comment]
https://user-images.githubusercontent.com/39250588/177781124-fa7183ad-efa9-474b-ab86-66250e82c63d.mp4

Maybe the problem is that it is set up for the Japan region.
We have changed the region of the open project from jp to us.

**Review by willeastcott — APPROVED** [review]
Final edit pass done. Looks fantastic! ❤️

**Comment by yaustar** [comment]
Cool, it looks like that works!

I did notice that the complete project https://playcanvas.com/project/926999/overview/playcanvas-photon-realtime still has some classes used for the pc scripts etc eg https://playcanvas.com/editor/code/926999?tabs=82205460

Can this be updated to use ES5 style that is done in the tutorial please? After that, we are good to go :)

**Comment by yushimatenjin** [comment]
Thank you!
> Can this be updated to use ES5 style that is done in the tutorial please? 
Yes, I will fix it as soon as possible!

**Comment by yushimatenjin** [comment]
Fixed !
- classes to prototype

> Cool, it looks like that works!
> 
> I did notice that the complete project https://playcanvas.com/project/926999/overview/playcanvas-photon-realtime still has some classes used for the pc scripts etc eg https://playcanvas.com/editor/code/926999?tabs=82205460
> 
> Can this be updated to use ES5 style that is done in the tutorial please? After that, we are good to go :)

I fixed it!
- all class to prototype

**Comment by yaustar** [comment]
Awesome, I get this merged and published :) Many thanks!

**Pull request merged**